java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java

```diff
@@ -330,7 +331,7 @@ public void attachFilter(ContentFilter filter) {
         if (idx != -1) {
             ContentProjectFilter toUpdate = filters.get(idx);
             if (toUpdate.getState() == ContentProjectFilter.State.DETACHED) {
-                toUpdate.setState(ContentProjectFilter.State.BUILT);
+                toUpdate.setState(ContentProjectFilter.State.EDITED);
```

Comment:
> This is not entirely correct. What if the filter wasn't edited at all? OTOH if we wanted to do this properly, we'd need to track the previous state of the object :-/     @hustodemon

Reply:
@hustodemon indeed, I tried to find the best solution with the information we had available. We need to discuss this.

Reply:
Or if we have the Built Time stored we can infer which state the filter should transit to comparing the dates. I think there it a task to implement that

Reply:
This is now fixed comparing the filter modified date with the environment built time